src/core/file.cc

```diff
@@ -619,13 +619,15 @@ static bool blockdev_nowait_works(dev_t device_id) {
     return blockdev_gen_nowait_works;
 }
 
-blockdev_file_impl::blockdev_file_impl(int fd, open_flags f, file_open_options options, dev_t device_id, size_t block_size)
+blockdev_file_impl::blockdev_file_impl(int fd, open_flags f, file_open_options options, dev_t device_id, size_t read_block_size, size_t write_block_size)
         : posix_file_impl(fd, f, options, device_id, blockdev_nowait_works(device_id)) {
-    // Configure DMA alignment requirements based on block device block size
-    _memory_dma_alignment = block_size;
-    _disk_read_dma_alignment = block_size;
-    _disk_write_dma_alignment = block_size;
-    _disk_overwrite_dma_alignment = block_size;
+    // Configure DMA alignment requirements based on block device characteristics
+    // - Read alignment: logical_block_size (no performance penalty for reading 512-byte sectors)
+    // - Write alignment: physical_block_size (avoids hardware-level RMW)
+    _memory_dma_alignment = write_block_size;
```

Comment:
memory_dma_alignment is fixed 512 IIRC regardless of logical/physical sector sizes.

Reply:
I see it can be obtained via statx stx_dio_mem_align.

Reply:
Interestingly it returns 4 for my nvme (and 512 for xfs files)